src/libraries/System.Private.CoreLib/src/System/Number.Formatting.cs

```diff
@@ -1080,6 +954,20 @@ private static unsafe void Int32ToNumber(int value, ref NumberBuffer number)
             number.CheckConsistency();
         }
 
+        public static string Int32ToDecStr(int value, int digits, IFormatProvider? provider)
```

Comment:
Would it be better to make this just:
```
public static string Int32ToDecStr(int value)
{
    return (value >= 0) ? UInt32ToDecStr((uint)value, -1) : NegativeInt32ToDecStr(value, -1, NumberFormatInfo.CurrentInfo.NegativeSign);
}
```
And inline the longer variant that takes `digits` and `provider` in `Int32Fmt.ToDecStr` only?

The advantage of this is that the inline callsite for e.g. `1.ToString()` would have smaller code and the method would be faster.

Reply:
It looks like the codegen would be the same, since the call to `GetInstance(IFormatProvider? formatProvider)` would be replaced with a direct call to `CurrentInfo` when `formatProvider` is `null`.

Reply:
I meant something like this: https://github.com/jkotas/runtime/commit/3fc231cdb8d33db7d1fd64b4a7907619bb0327f1

For example, it will reduce code size generated for `static string m(int a, int b, int c) => a.ToString() + b.ToString() + c.ToString();` from 0x54 bytes to 0x3c bytes (Windows x64).